runtime/function.c

```diff
@@ -0,0 +1,59 @@
+#include "balrt.h"
+
+bool _bal_function_subtype_contains(UniformSubtypePtr stp, TaggedPtr p) {
+    if ((getTag(p) & UT_MASK) != TAG_FUNCTION) {
+        return false;
+    }
+    FunctionValuePtr valPtr = taggedToPtr(p);
+    FunctionSignaturePtr signaturePtr = valPtr->signature;
+    MemberType actualReturnTy = signaturePtr->returnTy;
+    FunctionSubtypePtr fstp = (FunctionSubtypePtr)stp;
+    if (!memberTypeIsSubtypeSimple(actualReturnTy, fstp->returnBitSet)) {
+        return false;
+    }
+    int64_t nParams = signaturePtr->nParams;
+    for (int64_t i = 0; i < nParams; i++) {
+        MemberType paramTy = signaturePtr->paramTys[i];
+        uint32_t paramBitSet;
+        if (i > fstp->nParams) {
+            paramBitSet = fstp->restBitSet;
+        }
+        else {
+            paramBitSet = fstp->paramBitSets[i];
+        }
+        if (!memberTypeIsSubtypeSimple(paramTy, paramBitSet)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool _bal_is_exact(FunctionSignaturePtr signature, FunctionValuePtr value) {
```

Comment:
should this be named `_bal_function_is_exact`?

Reply:
fixed